services/QuillLMS/app/services/analytics/segment_analytics.rb

```diff
@@ -148,6 +148,35 @@ def track_previewed_activity(user_id, activity_id)
 
   end
 
+  def trigger_teacher_subscription_will_expire(subscription_id)
+    subscription = Subscription.find(subscription_id)
+    teacher_id = subscription.users.first.id
+
+    track({
+      user_id: teacher_id,
+      event: SegmentIo::BackgroundEvents::TEACHER_SUB_WILL_EXPIRE,
+      properties: {
+        subscription_id: subscription.id
+      }
+    })
+  end
+
+  def trigger_school_subscription_will_expire(subscription_id)
+    subscription = Subscription.find(subscription_id)
+    school_id = subscription.schools.first.id
+
+    track({
+      # Segment requires us to send a unique User ID or Anonymous ID for every event
+      # generate a random UUID here because we don't want the School Subscription event to be associated to any real user
```

Comment:
We have the `purchaser_id` on the subscription itself - maybe it makes sense to use that where we have it? It isn't present on all subscriptions but it seems like it might be useful when it is?

Reply:
That makes sense, I'll add this field as user ID when we have the value.